src/App.js

```diff
@@ -21,38 +21,112 @@ const generateSquares = () => {
       currentId += 1;
     }
   }
-
+  
   return squares;
 }
 
-const App = () => {
 
-  const [squares, setSquares] = useState(generateSquares());
+const App = () => {
 
+  const blankBoard = generateSquares();
+  const [squares, setSquares] = useState(blankBoard); 
+  const [currentPlayer, setCurrentPlayer] = useState(PLAYER_1); 
+  const [numOfSquareOccupied, setNumOfSquareOccupied] = useState(1);
+  const [currentWinner, setCurrentWinner] = useState(null);
+  const [status, setStatus] = useState('Current player is X'); 
```

Comment:
Do you really need all these as state?  I think you could derive status from `currentPlayer`.